src/dachshund/candidate.rs

```diff
@@ -76,7 +80,8 @@ where
     max_core_node_edges: usize,
     ties_between_nodes: usize,
     local_guarantee: LocalDensityGuarantee,
-    neighborhood: HashMap<NodeId, usize>,
+    neighborhood: Option<HashMap<NodeId, usize>>,
+    recipe: Option<(Option<u64>, NodeId)>,
```

Comment:
I've never been crazy about using unnamed tuples for anything, much less function parameters. It's SO easy to create at minimum a new type, if not an actual struct with field names (which is what I prefer).

However, the code base already has a ton of such tuples in it, I foresee either me just getting over this, or else doing a single replacement sweep of the whole project sometime.

Reply:
I agree. I don't want more of these to accumulate. I'll switch this to a struct.